adding requestHeaders to the websocket client

manute opened the pull request:
Adding `requestHeaders` argument to the websocket client funcs to have the possibilty of passing request Headers .

For example, if you need to pass other authorization header like `X-Auth-Token` or/and  other headers like `Sec-WebSocket-Key` , `Sec-WebSocket-Version`this PR opens that possibility.

Example: 
```golang
client.Websocket(query, http.Header{
     "X-Auth-Token": {"validToken"},
     "Sec-WebSocket-Key": {"SGVsbG8sIHdvcmxkIQ=="},
     "Sec-WebSocket-Version": {13},
}
```

manute commented:
hmm the lint fails with 
```
client/websocket.go:56:46: bodyclose: response body must be closed (bodyclose)
	c, resp, err := websocket.DefaultDialer.Dial(url, reqHeaders)
``` 
but that body is being closed inside the `Subscriptions.Close` func  and I've not touched any code there 🤔

vektah commented:
> hmm the lint fails with

try a rebase, I fixed a few of these ~0.9.2 ish.

vektah commented:
PR message could use more explaining why this would be useful, and probably tests/docs on showing how it would be used.

manute commented:
> try a rebase, I fixed a few of these ~0.9.2 ish.

I did and the same results 😢 .

> PR message could use more explaining

Added. Let  me know if more doc or tests are needed , although the type added is  pretty clear

vektah commented:
Hmm yeah, it's not failing on master but something about your change is confusing the linter.

Lets just skip it, this is just the test client.
```
c, resp, err := websocket.DefaultDialer.Dial(url, reqHeaders) //nolint:bodyclose
```

vektah commented:
Please a look at https://github.com/99designs/gqlgen/pull/861 and see if it covers your use case

manute commented:
yeah I think with https://github.com/99designs/gqlgen/pull/861/files#diff-3c0b8ea9d09776a0db9bab04cf0a6ea7R32 it can be added the headers to the websocket requests. 

 I'll close the issue in lieu of that PR.
Thanks @vektah ,
